src/electronApp/core/dataSaver.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Save data locally
+ */
+export class DataSaver {
+  static readonly lastIPsKey = "lastIPs";
+
+  /**
+   * Saves the IP-Address into the list of last used IPs
+   * @param ip The ip address to be saved
+   */
+  public static SaveLastIP(ip: string): void {
+    let LastIPs = localStorage.getItem(this.lastIPsKey);
+    
+    if (LastIPs != null) { // If there are already IPs saved
+      let IPs: string[] = LastIPs.toString().split(",");
+      let index = IPs.indexOf(ip);
+
+      if (index != -1) { // If the IP is already in the list, move it to first place
+        this.array_move(IPs, index, 0);
+      } 
+      else { // If it's not in the list add it and put it in first place
+        let length = IPs.push(ip);
+        this.array_move(IPs, length - 1, 0);
+      }
+      
+      localStorage.setItem(this.lastIPsKey, IPs.toString());
+    } else { // If there are no IPs saved
+      let IPs: string[];
+      IPs = [ip];
+      localStorage.setItem(this.lastIPsKey, IPs.toString());
+    }
+  }
+
+  /**
+   * Swaps the elements at old_index and new_index
+   * @param arr Array in which the elements are being moved
+   * @param old_index The index of the element to be moved
+   * @param new_index The index of the position the element will be moved to
+   */
+  static array_move(arr, old_index, new_index): void {
```

Comment:
Guessing this was borrowed from here: https://stackoverflow.com/questions/5306680/move-an-array-element-from-one-array-position-to-another

It seems a lite more complex than it needs to be for what it is doing. I don't think it needs to deal with the case of the move index being off the end of the array, indeed this would be an error for the UI, so it would have been simpler to support move & validate the input is valid.